### ISSUE TITLE
proposal: add option to  enable tunning socket listen backlog

### ISSUE THREAD

**Issue by sillyousu** [mgravell/Pipelines.Sockets.Unofficial#29]
```diff
--- a/src/Pipelines.Sockets.Unofficial/SocketServer.cs
+++ b/src/Pipelines.Sockets.Unofficial/SocketServer.cs
 using System.IO.Pipelines;
 using System.Net;
@@ -22,6 +22,7 @@ namespace Pipelines.Sockets.Unofficial
             AddressFamily addressFamily = AddressFamily.InterNetwork,
             SocketType socketType = SocketType.Stream,
             ProtocolType protocolType = ProtocolType.Tcp,
+            int listenBacklog = 20,
             PipeOptions sendOptions = null, PipeOptions receiveOptions = null)
         {
             if (_listener != null) Throw.InvalidOperation("Server is already running");
```

**Comment by Salty-Sailor** [comment]
:+1:

**Comment by mgravell** [comment]
adding cross-reference: https://github.com/mgravell/Pipelines.Sockets.Unofficial/pull/30

**Comment by mgravell** [comment]
as mentioned in the PR feedback: fine with the concept, but needs to be done in a binary-compatible way; specifics added in PR discussion.

**Comment by mgravell** [comment]
available in anything from 2.0.11
